### ISSUE TITLE
How can I run all features at same on one port using single command? 

### ISSUE THREAD

**Issue by mishra011** [RasaHQ/rasalit#58]

**Comment by koaning** [comment]
You'd need to run something like `nginx` in front of it as a proxy. This is possible, but not super obvious to get right.

**Comment by mishra011** [comment]
@koaning Yes, we can do it, we can run all of them separately on different ports then use a reverse proxy to combine them on one port. But I was wishing since most people will be using all of them, so instead of running all 4 separately, shouldn't we have a feature to run all of them at once then use them accordingly.

**Comment by koaning** [comment]
You're the first person suggesting to use all views at the same time actually. I'm open to a tutorial for the readme but I doubt streamlit open aohrce allows you to spin up multiple apps on a single port.

**Comment by mishra011** [comment]
I think all views at the same place can be a nice feature if streamlit supports it.

**Comment by drewsilcock** [comment]
I agree this would be a very useful feature to make the tool more user-friendly and easier to deploy for general use rather than individual developers or data scientists playing around with it.

In the meantime it would be helpful to have a `Dockerfile` with an image for each tool, that way you could easily spin all of the tools at the same time along with an nginx reverse proxy using Docker Compose, without needing to change any Python code.

**Comment by koaning** [comment]
I'll gladly review any PR for a Dockerfile that does this.

**Comment by koaning** [comment]
This might also make it easy to ship the bulk labelling tool now that I think of it. Althought probably best to keep as a separate Dockerfile.

**Comment by drewsilcock** [comment]
It would also be possible to implement a sidebar navigation like this other repo - [awesome-streamlit](https://github.com/MarcSkovMadsen/awesome-streamlit).

See: https://github.com/MarcSkovMadsen/awesome-streamlit/blob/be454e29c35a9a1a760b1737a5176a47f4f9717b/app.py#L5-L30

**Comment by drewsilcock** [comment]
Pushed a PR containing a `Dockerfile`, `docker-compose.yaml` and Nginx configuration for this reverse proxying solution: #60.
